core/src/main/scala/akka/kafka/ProducerSettings.scala

```diff
@@ -273,7 +273,7 @@ class ProducerSettings[K, V] @InternalApi private[kafka] (
    * Replaces the default Kafka producer creation logic.
    */
   @InternalApi private[kafka] def withProducerFactory(
```

Comment:
As @charlibot suggests, you may remove the Internal API flags to open up for other use cases.

Reply:
I removed the tag internal API